### PR TITLE
move to mkdocs

### PR DESCRIPTION
**Pull request by dholbach**
Following some of our other projects, looking at `mkdocs` for `wksctl` too.

To test it, run

```
pip3 install -r docs/requirements.txt
mkdocs serve
```

I can point http://wksctl.rtfd.io/ to use this branch for testing too.

### CODE REVIEW

**Comment by dholbach** [comment]
Suggested new docs are: https://test-wksctl.readthedocs.io/

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+site_name: wksctl
+# site_description:
```
Comment (fbarl) [inline mkdocs.yml]: Maybe we could fill this one out

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+site_url: https://wksctl.readthedocs.io
+
+# Repository
+repo_name: weaveworks/wksctl
+repo_url: https://github.com/weaveworks/wksctl
+edit_uri: ""
+
+theme:
+  name: material
+  #  logo: logo.png
```
Comment (fbarl) [inline mkdocs.yml]: Could we use a Weaveworks logo here? Or is that not desirable? cc @bia

File: docs/_static/custom.css [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
 
-code {
-  font-family: "Roboto Mono", monospace;
+.md-header, .md-footer-nav {
+    background-image: linear-gradient(45deg, rgb(0, 150, 225) 0%, rgb(27, 141, 226) 24%, rgb(42, 125, 227) 53%, rgb(53, 112, 227) 78%, rgb(53, 112, 227) 100%);
 }
 
+.md-header-nav__title {
+    font-size: .85rem;
+}
```
Comment (fbarl) [inline docs/_static/custom.css]: Where do these overrides come from?

File: .github/workflows/check-links.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+on: [push, pull_request]
```
Comment (fbarl) [inline .github/workflows/check-links.yaml]: Maybe add a short comment explaining what this workflow does and why (also where it's been taken from) :)

File: Makefile [hunk 5]
```diff
@@ -147,12 +145,3 @@ integration-tests-container: cmd/wksctl/wksctl cmd/controller/.uptodate
...
 
-build-docs:
-	@cd docs && docker build -t wksctl-docs .
-
-test-docs: build-docs
-	@docker run -it wksctl-docs /usr/bin/linkchecker _build/html/index.html
-
-serve-docs: build-docs
-	@echo Stating docs website on http://localhost:${DOCS_PORT}/_build/html/index.html
-	@docker run -i -p ${DOCS_PORT}:8000 -e USER_ID=$$UID wksctl-docs
```
Comment (fbarl) [inline Makefile]: Is it still possible to run a local docs server now with `mkdocs`? Maybe it'd be good to have some sort of a make script for local testing.

Comment (fbarl) [reply Makefile]: Ah yes, I now saw this:

```
pip3 install -r docs/requirements.txt
mkdocs serve
```

So how about putting those under `make serve-docs`? :)

File: docs/_static/custom.css (hunk 3 above)
Comment (dholbach) [reply docs/_static/custom.css]: @hiddeco and I took inspiration from https://github.com/openfaas/docs/blob/master/mkdocs.yml and https://github.com/openfaas/docs/blob/master/docs/stylesheets/extra.css and decided in https://github.com/fluxcd/flux/pull/2919#issuecomment-600177513 to use the above.

**Comment by bia** [comment]
Looks great! Thanks @dholbach!

What do you guys think about making the navbar white to match [eksctl](http://mhausenblas.info/eksctl/)?

**Comment by dholbach** [comment]
I personally feel the all-white is a bit dull. Could we use other colours that make sense? What does everybody else think?

**Comment by dholbach** [comment]
I stole the navbar colours and made a variation of it after reading https://github.com/openfaas/docs/blob/master/mkdocs.yml and https://github.com/openfaas/docs/blob/master/docs/stylesheets/extra.css

**Comment by bia** [comment]
Your version is more beautiful. Let's just keep it unchanged and merge. :)

**Comment by dholbach** [comment]
Yes. Let's do any clean up and beautification in a new PR. Thanks for the reviews @bia and @fbarl ! <3

**Comment by dholbach** [comment]
I deleted `test-wksctl.rtfd.io`.

**Pull request merged**